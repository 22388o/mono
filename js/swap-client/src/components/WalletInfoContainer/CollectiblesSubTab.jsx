import { useEffect, useSyncExternalStore } from "react";

import { Divider, Stack, Typography } from "@mui/material"
import KeyboardArrowRightIcon from '@mui/icons-material/KeyboardArrowRight';

import styles from "../../styles/WalletInfoContainer.module.css";
import { walletStore } from "../../syncstore/walletstore";
import { getBTCPrice, getETHPrice } from "../../utils/apis";


export const CollectiblesSubTab = () => {
  const globalWallet = useSyncExternalStore(walletStore.subscribe, () => walletStore.currentState)
  const assets = globalWallet.assets;
  const coin_prices = globalWallet.coin_prices;

  useEffect(() => {
    const core = async () => {
      const btc = await getBTCPrice();
      const eth = await getETHPrice();
      walletStore.dispatch({ type: 'SET_COIN_PRICES', payload: {
        BTC: btc,
        ETH: eth,
        fetching: false
      }});
    };
    core();
  }, []);

  const renderItem = (asset) => {
    return (
      <Stack direction='row' className={styles['wallet-collectible-item']}>
        <Stack direction='row' gap={1} sx={{alignItems:'center', marginTop: '2px'}}>
          <img width={45} src={asset.img_url} />
          <Typography className={styles['wallet-item-title']}>{asset.title}</Typography>
        </Stack>
        <Stack direction='row' gap={1} sx={{alignItems:'center'}}>
          <Stack direction='row' sx={{alignItems:'flex-end'}} gap={0.5}>
            <Typography sx={{fontFamily:'NotoBold', color:'white', fontSize:'20px'}}>{ asset.balance }</Typography>
            <Typography sx={{color:'#AAAAAA', marginBottom: '1px'}}>{ asset.type }</Typography>
          </Stack>
          <KeyboardArrowRightIcon />
        </Stack>
      </Stack>
    )
  }

  return (
    <Stack>
      { assets.filter(asset => asset.isNFT === true).map((asset, idx) => 
        <div key={idx}>
          { renderItem(asset) }
          { <Divider /> }
        </div>
      ) }
    </Stack>
  )
}