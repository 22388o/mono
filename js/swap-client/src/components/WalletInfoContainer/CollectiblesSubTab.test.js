import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom'; // For the "toBeInTheDocument" matcher
import { CollectiblesSubTab } from './CollectiblesSubTab';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ data: { amount: 12 } }),
  })
);


describe('CollectiblesSubTab component test', () => {
  it('renders component', async () => {

    const {container} = render(
      <CollectiblesSubTab />
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    })
  });
});
