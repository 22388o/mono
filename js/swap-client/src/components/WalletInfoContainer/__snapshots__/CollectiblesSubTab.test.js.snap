// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectiblesSubTab component test renders component 1`] = `
<div>
  <div
    class="MuiStack-root css-nen11g-MuiStack-root"
  >
    <div>
      <div
        class="MuiStack-root css-m69qwo-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1tjhhxk-MuiStack-root"
        >
          <img
            src="/nft/1.png"
            width="45"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Inscription #0
          </p>
        </div>
        <div
          class="MuiStack-root css-1qxw92a-MuiStack-root"
        >
          <div
            class="MuiStack-root css-7vetex-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
            >
              1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
            >
              BTCORD-0
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
    <div>
      <div
        class="MuiStack-root css-m69qwo-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1tjhhxk-MuiStack-root"
        >
          <img
            src="/nft/2.png"
            width="45"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Inscription #1
          </p>
        </div>
        <div
          class="MuiStack-root css-1qxw92a-MuiStack-root"
        >
          <div
            class="MuiStack-root css-7vetex-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
            >
              1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
            >
              BTCORD-1
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
    <div>
      <div
        class="MuiStack-root css-m69qwo-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1tjhhxk-MuiStack-root"
        >
          <img
            src="/nft/3.png"
            width="45"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Inscription #2
          </p>
        </div>
        <div
          class="MuiStack-root css-1qxw92a-MuiStack-root"
        >
          <div
            class="MuiStack-root css-7vetex-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1fmxcpf-MuiTypography-root"
            >
              1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-xrfphg-MuiTypography-root"
            >
              BTCORD-2
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
  </div>
</div>
`;
